sklearn/cluster/tests/test_k_means.py

```diff
@@ -966,11 +1039,91 @@ def test_minibatch_kmeans_partial_fit_int_data():
     assert km.cluster_centers_.dtype.kind == "f"
 
 
-def test_result_of_kmeans_equal_in_diff_n_jobs():
+def test_result_of_kmeans_equal_in_diff_n_threads():
     # PR 9288
     rnd = np.random.RandomState(0)
     X = rnd.normal(size=(50, 10))
 
-    result_1 = KMeans(n_clusters=3, random_state=0, n_jobs=1).fit(X).labels_
-    result_2 = KMeans(n_clusters=3, random_state=0, n_jobs=2).fit(X).labels_
+    with threadpool_limits(limits=1, user_api="openmp"):
+        result_1 = KMeans(
+            n_clusters=3, random_state=0).fit(X).labels_
+    with threadpool_limits(limits=2, user_api="openmp"):
+        result_2 = KMeans(
+            n_clusters=3, random_state=0).fit(X).labels_
     assert_array_equal(result_1, result_2)
+
+
+@pytest.mark.parametrize("precompute_distances", ["auto", False, True])
+def test_precompute_distance_deprecated(precompute_distances):
+    # FIXME: remove in 0.25
+    depr_msg = ("'precompute_distances' was deprecated in version 0.23 and "
+                "will be removed in 0.25.")
+    X, _ = make_blobs(n_samples=10, n_features=2, centers=2, random_state=0)
+    kmeans = KMeans(n_clusters=2, n_init=1, init='random', random_state=0,
+                    precompute_distances=precompute_distances)
+
+    with pytest.warns(FutureWarning, match=depr_msg):
+        kmeans.fit(X)
+
+
+@pytest.mark.parametrize("n_jobs", [None, 1])
+def test_n_jobs_deprecated(n_jobs):
+    # FIXME: remove in 0.25
+    depr_msg = ("'n_jobs' was deprecated in version 0.23 and will be removed "
+                "in 0.25.")
+    X, _ = make_blobs(n_samples=10, n_features=2, centers=2, random_state=0)
+    kmeans = KMeans(n_clusters=2, n_init=1, init='random', random_state=0,
+                    n_jobs=n_jobs)
+
+    with pytest.warns(FutureWarning, match=depr_msg):
+        kmeans.fit(X)
+
+
+def test_warning_elkan_1_cluster():
+    X, _ = make_blobs(n_samples=10, n_features=2, centers=1, random_state=0)
+    kmeans = KMeans(n_clusters=1, n_init=1, init='random', random_state=0,
+                    algorithm='elkan')
+
+    with pytest.warns(RuntimeWarning,
+                      match="algorithm='elkan' doesn't make sense for a single"
+                            " cluster"):
+        kmeans.fit(X)
+
+
+def test_error_wrong_algorithm():
+    X, _ = make_blobs(n_samples=10, n_features=2, centers=2, random_state=0)
+    kmeans = KMeans(n_clusters=2, n_init=1, init='random', random_state=0,
+                    algorithm='wrong')
+
+    with pytest.raises(ValueError,
+                       match="Algorithm must be 'auto', 'full' or 'elkan'"):
+        kmeans.fit(X)
+
+
+@pytest.mark.parametrize("array_constr",
+                         [np.array, sp.csr_matrix],
+                         ids=['dense', 'sparse'])
+@pytest.mark.parametrize("algo", ['full', 'elkan'])
+def test_k_means_1_iteration(array_constr, algo):
+    # check k_means results for a single iteration (EME) vs pure python implem.
```

Comment:
```suggestion
    # check k_means results in a single iteration (EME) vs pure python implem.
```

Reply:
I reworded the comment to make it a bit clearer